osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/UpdatePortGroupTaskTest.java

```diff
@@ -0,0 +1,318 @@
+
+		NetworkElement ne = createNetworkElement("openstackId");
+
+		SdnRedirectionApi redirectionApi = mockRegisterNetworkElement(portGroup, ne);
+		registerNetworkRedirectionApi(redirectionApi, sg.getVirtualizationConnector());
+
+		UpdatePortGroupTask task = new UpdatePortGroupTask(sg, portGroup);
+
+		// Act.
+		task.execute();
+
+		// Assert.
+		verify(this.em, Mockito.never()).merge(any());
+	}
+
+	@Test
+	public void testExecute_UpdatePortGroupNeededWithDifferentElementId_ExecutionFinishes() throws Exception {
+
+		// Arrange.
+		SecurityGroup sg = createSecurityGroup(1L, "tenantId", "tenantName", 1L, "sgName");
+		sg.addSecurityGroupMember(newSGMWithPort(VM.class, 1L, "openstackId"));
+		PortGroup portGroup = createPortGroup("different_openstackId");
+
+		List<NetworkElement> neList = ostRegisterPorts(sg);
+
+		String domainId = UUID.randomUUID().toString();
+		registerDomain(domainId, sg);
+
+		NetworkElement ne = createNetworkElement("openstackId");
+
+		SdnRedirectionApi redirectionApi = mockRegisterNetworkElement(portGroup, ne);
+		registerNetworkRedirectionApi(redirectionApi, sg.getVirtualizationConnector());
+
+		UpdatePortGroupTask task = new UpdatePortGroupTask(sg, portGroup);
+
+		// Act.
+		task.execute();
+
+		// Assert.
+		verify(this.em, Mockito.times(1)).merge(sg);
+		Assert.assertEquals(ne.getElementId(), sg.getNetworkElementId());
+		assertNetworkElementsParentIdWithDomainId(neList, domainId);
+	}
+
+	@Test
+	public void testExecute_WhenDomainIsNotFound_ThrowsException() throws Exception {
+
+		// Arrange.
+		SecurityGroup sg = createSecurityGroup(1L, "tenantId", "tenantName", 1L, "sgName");
+		sg.addSecurityGroupMember(newSGMWithPort(VM.class, 1L, "openstackId"));
+		PortGroup portGroup = createPortGroup("openstackId");
+
+		ostRegisterPorts(sg);
+
+		// domain id null
+		String domainId = null;
+		registerDomain(domainId, sg);
+
+		this.exception.expect(Exception.class);
+		this.exception.expectMessage(
+				String.format("Failed to retrieve domainId for given tenant: '%s' and Security Group: '%s",
+						sg.getTenantName(), sg.getName()));
+
+		UpdatePortGroupTask task = new UpdatePortGroupTask(sg, portGroup);
+
+		// Act.
+		task.execute();
+
+		// Assert.
+		verify(this.em, Mockito.never()).merge(any());
+	}
+
+	@Test
+	public void testExecute_WhenPortGpIsNotFound_ThrowsException() throws Exception {
+
+		// Arrange.
+		SecurityGroup sg = createSecurityGroup(1L, "tenantId", "tenantName", 1L, "sgName");
+		sg.addSecurityGroupMember(newSGMWithPort(VM.class, 1L, ""));
+		PortGroup portGroup = createPortGroup("openstackId");
+
+		ostRegisterPorts(sg);
+
+		String domainId = UUID.randomUUID().toString();
+		registerDomain(domainId, sg);
+
+		// network element/portGp null
+		NetworkElement ne = null;
+		SdnRedirectionApi redirectionApi = mockRegisterNetworkElement(portGroup, ne);
+		registerNetworkRedirectionApi(redirectionApi, sg.getVirtualizationConnector());
+
+		this.exception.expect(Exception.class);
```

Comment:
Please put a TODO in the code that is throwing `Exception` so we can address it later. It should throw a more specific exception.